### ISSUE TITLE
Add time format customization example

### ISSUE THREAD

**Issue by taldcroft** [astropy/astropy#4631]
In the astropy mailing list I posted an example of customizing the style of the yday format.  This should go in the docs as this might be one of the more common use cases for customizing (i.e. minor twiddles in formatting):
```
class TimeYearDayTimeCustom(TimeISO):
  """
  Year, day-of-year and time as "<YYYY>-<DOY>T<HH>:<MM>:<SS.sss...>".
  The day-of-year (DOY) goes from 001 to 365 (366 in leap years).
  For example, 2000-001T00:00:00.000 is midnight on January 1, 2000.
  The allowed subformats are:
  - 'date_hms': date + hours, mins, secs (and optional fractional secs)
  - 'date_hm': date + hours, mins
  - 'date': date
  """

  name = 'yday_custom'
  subfmts = (('date_hms',
              '%Y-%jT%H:%M:%S',
              '{year:d}-{yday:03d}T{hour:02d}:{min:02d}:{sec:02d}'),
             ('date_hm',
              '%Y-%jT%H:%M',
              '{year:d}-{yday:03d}T{hour:02d}:{min:02d}'),
             ('date',
              '%Y-%j',
              '{year:d}-{yday:03d}'))

In [4]: t = Time.now()

In [5]: t.yday_custom
Out[5]: '2016-053T12:01:41.663'

In [6]: t2 = Time('2016-001T00:00:00')

In [7]: t2.iso
Out[7]: '2016-01-01 00:00:00.000'
```

**Comment by pllim** [comment]
:+1:

Does this *replace* the current doc, or *in addition* to it?

**Comment by taldcroft** [comment]
In addition to, but most likely *before* the existing examples.

**Comment by pllim** [comment]
Sounds good!

**Comment by goelakash** [comment]
Hi, I'm new here and would like to contribute.
The signature `class TimeYearDayTimeCustom(TimeISO)` doesnt seem to be in  `time/__init__.py`. Does this example add to `class TimeDecimalYear(TimeFormat)` ?

**Comment by pllim** [comment]
@goelakash , it is simply a documentation update. There is no need to change any Python codes, just the RST file.

**Comment by goelakash** [comment]
There seems to be no `TimeYearDayTimeCustom` in the rst file (doc/time/index.rst) either.

**Comment by pllim** [comment]
@goelakash , that is the point. It is not there yet. We need someone to add it. :)

**Comment by goelakash** [comment]
:+1:

**Comment by mandarGG** [comment]
I would like to contribute to this issue.I have understood that  time format customization is a thing not added in the documentation and that is to be done(Please correct me if  i am wrong). Can anyone help me out with this and how do i start?

**Comment by taldcroft** [comment]
@mandarGG - yes, just add to the sphinx docs.  You can start by reading the developer docs: http://astropy.readthedocs.org/en/latest/index.html#developer-documentation.

**Comment by mandarGG** [comment]
I read the docs on Time in astropy and found that when we call the Time.now() function it creates an instance ( a new Time object) which contains current time in multiple formats like jd and also that yday_custom is one such time format allowing to represent current time in 3 subformats .I have to add the TimeYearDayTimeCustom class in the documentation mentioning its use as mentioned above .But how do i get the time in each of the 3 subformats  mentioned here.Also how do i start to add this thing to the docs? Please guide and correct me if i am wrong

**Comment by taldcroft** [comment]
@mandarGG - yes, add to `doc/time/index.rst`.  You can also check the astropy mailing list for the original discussion to put things in context.  Apart from that you should be able to find answers to your questions in the astropy docs.

**Comment by mandarGG** [comment]
I have included the necessary documentation by modifying /doc/time/index.rst. I have added the time format customization after description of all the time formats. I thought  it would be better to upload the file here individually and modify it according to the reviews.Also if the modified documentation is fine, then what should i do next? Please correct me if i am wrong.
[Modified file.zip](https://github.com/astropy/astropy/files/178513/Modified.file.zip)

**Comment by pllim** [comment]
@mandarGG , instead of a ZIP file, you should file a PR.

**Comment by mandarGG** [comment]
oh ok so i should do i pull request on github? Will do it now

**Comment by mandarGG** [comment]
On running the code mentioned above by adding :from astropy.time import * at the top(above the class definition) i am getting the error 'Time' object has no attribute 'yday_custom'.This is the same reason why my checks have been failing.What is the problem here ? Initially i was getting TimeISO not defined, then i saw the docs and realised that there is a class called astropy.time.TimeISO so i rectified the error. @pllim  has said in the above comments that there is no need to change any actual code.So if yday_custom is absent in the time formats, then how will the given statements return an output which is the error i am facing. Am i right in my understanding? Please correct me if i am wrong

**Comment by pllim** [comment]
I was under the impression that you actually have to define the `TimeYearDayTimeCustom(TimeISO)` class yourself *and then* use it. But I do not see that class definition in #4703.

**Comment by SaraOgaz** [comment]
Who can close this issue?  Only @taldcroft?

**Comment by pllim** [comment]
I can :smile:

**Comment by pllim** [comment]
Fixed by #4902
